lib/java_buildpack/framework/contrast_security_agent.rb

```diff
@@ -49,7 +49,13 @@ def release
 
       # (see JavaBuildpack::Component::VersionedDependencyComponent#jar_name)
       def jar_name
-        "contrast-engine-#{@version.to_s.split('_')[0]}.jar"
+        boot_version = @version.to_s.split('_')[0]
+        major, minor, patch = boot_version.split('.').map(&:to_i)
```

Comment:
No need to do this split and comparison, the `TokenizedVersion` is comparable.  [Example](https://github.com/cloudfoundry/java-buildpack/blob/3b0fa53b5ad50ded173f73ec95e7c517fb9f60bd/lib/java_buildpack/component/mutable_java_home.rb#L39)